### DISCUSSION TITLE
blendplot

### DISCUSSION THREAD

**Discussion by Tillsten** [pyqtgraph/pyqtgraph#2190] [Ideas]
Hey,
i made a small toy plotting library for qt using Blend2D as a renderer:

https://github.com/Tillsten/blendplotpy

Hence rendering is still happing on the CPU. The performance is quite fantastic, especially when it comes to lines with larger linewidths. Maybe it is worth to check out if it feasible to use Blend2d for pyqtgraph, even just doing the line rendering in Blend2d should give a nice boost.

Currently it uses cppyy to interface with cpp interface of blend2d, but this could be replaced with pybind11 during build time. It also uses a numba-accelrated version of the tick-selection algorithm implemented in vispy (vendored), which also not really necessary.

Any comments appriciated,
Till

**Comment by Tillsten** [comment]
Example video in bad quality:

https://user-images.githubusercontent.com/189880/152535030-c441a694-284c-4f07-a081-be5e5a923a84.mp4

**Reply by danielhrisca** [reply]
One of the main reasons people use pyqtgraph is the features it provides. But sometimes speed is also required so faster plotting is always nice

How is it handling bigger curve count? For example 1000 curves with 2000 points per curve

Can you have separate zooming per curve? In pyqtgraph one can have one View box per curve so that y zooming can be done for each curve individually
